dds/DCPS/QueryConditionImpl.h

```diff
@@ -51,10 +51,19 @@ class OpenDDS_Dcps_Export QueryConditionImpl
 
   bool hasFilter() const;
 
+  /**
+   * Returns true if the sample matches the query.
+   */
   template<typename Sample>
-  bool filter(const Sample& s) const
+  bool filter(const Sample& s, bool has_invalid_data) const
   {
     ACE_GUARD_RETURN(ACE_Recursive_Thread_Mutex, guard, lock_, false);
+    const MetaStruct& meta = getMetaStruct<Sample>();
+    // Omit the sample from results if there are non-key fields in the query
+    // and the query only has key fields.
```

Comment:
Shouldn't this say something like "and the received sample only has key fields."?

Reply:
Yes, it was a typo and I will push a correction.